File: engine/grid/los.py
"""Line of sight helpers."""

from __future__ import annotations

from engine.core.state import BattleState, UnitState
from engine.grid.loe import has_line_of_effect


def has_line_of_sight(state: BattleState, source: UnitState, target: UnitState) -> bool:
    return has_line_of_effect(state, source, target)

File: old/src/grid/los.ts
/**
 * Line of sight helpers.
 * Mirrors engine/grid/los.py
 */

import { BattleState, UnitState } from "../engine/state";
import { hasLineOfEffect } from "./loe";

export function hasLineOfSight(
  state: BattleState,
  source: UnitState,
  target: UnitState,
): boolean {
  return hasLineOfEffect(state, source, target);
}
